calico-vpp-agent/services/service_server.go

```diff
@@ -319,7 +326,12 @@ func (s *Server) Serve() {
 
 	s.t.Go(func() error { s.serviceInformer.Run(s.t.Dying()); return nil })
 	s.t.Go(func() error { s.endpointInformer.Run(s.t.Dying()); return nil })
-	<-s.t.Dying()
+	select {
```

Comment:
If we receive a Dying event from the main Tomb, we want to stop the 2 informers running in `s.t`, not just exit.